### DISCUSSION TITLE
Enemy AI using ranged weapon can't properly deal with character on ladder

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15373] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

in the video I tested monster AI (Piratehuman09) and human AI (Team2 security officer)
they can't properly attack me when I am on the ladder
monster AI always tends to climb the ladder first while they can just attack me on the ground
human AI will attack me when I am within his sight then climb the ladder
but they have same issue: unable to stop and attack me on the ladder

https://github.com/user-attachments/assets/6a86f21c-543c-48d1-b123-d2582b1c64ff

### Reproduction steps

.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Minedeployder** [comment]
very cool

**Comment by SoobGorl** [comment]
Related to [#16333](https://github.com/FakeFishGames/Barotrauma/discussions/16333)

**Comment by SoobGorl** [comment]
Seems to have been a bit improved since this ticket was made--enemy was able to attack when repeating these steps.

https://github.com/user-attachments/assets/04f1d6b7-9416-4d24-8ae3-1e3f39968a8f

More powerful weapon wielders (like SMGs) may still struggle, but wasn't able to test this well due to commands in the video changing (and couldn't think of any standard bots to test with that meet the criteria besides the captain).

**Comment by SoobGorl** [comment]
Pieced together a way of doing it proper--looks to be fixed/working accordingly in the latest internal branch.


https://github.com/user-attachments/assets/625278c2-5457-4581-968e-754cef220ed8
